test/todo_cubit_test.dart

```diff
@@ -0,0 +1,170 @@
+import 'package:bloc_test/bloc_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:objectbox/objectbox.dart';
+import 'package:reactive_todo_app/features/todo/data/model/todo.dart';
+import 'package:reactive_todo_app/features/todo/presentation/cubit/todo_cubit.dart';
+import 'package:reactive_todo_app/features/todo/presentation/cubit/todo_state.dart';
+import 'package:reactive_todo_app/main.dart';
+
+void main() {
+  setUp(() async {
+    WidgetsFlutterBinding.ensureInitialized();
+    final Store store = FakeObjectBoxStore();
+    await registerFactories(store: store);
+    sl.allowReassignment = true;
+    sl.registerFactory<Box<Todo>>(() => FakeTodoBox());
+  });
+
+  test('cubit initial state should be empty', () {
+    expect(sl<TodoCubit>().state, equals(TodoInitialState()));
+  });
+
+  group('get todo', () {
+    blocTest(
+      'cubit emits TodoEmptyState when todo box is empty',
+      setUp: () => mockTodoBoxResult(results: []),
+      build: () => sl<TodoCubit>(),
+      act: (TodoCubit cubit) => cubit.getTodos(),
+      expect: () => [
+        TodoEmptyState(),
+      ],
+    );
+
+    final List<Todo> todos = [
+      Todo('Pray'),
+      Todo('Eat'),
+      Todo('Work'),
+    ];
+
+    blocTest(
+      'cubit emits TodoLoadedState when todo box is not empty',
+      setUp: () => mockTodoBoxResult(results: todos),
+      build: () => sl<TodoCubit>(),
+      act: (TodoCubit cubit) => cubit.getTodos(),
+      expect: () => [
+        TodoLoadedState(todos),
+      ],
+    );
+
+    blocTest(
+      'cubit emits TodoLoadedState when search with known keyword',
+      setUp: () => mockTodoBoxResult(results: todos),
+      build: () => sl<TodoCubit>(),
+      act: (TodoCubit cubit) => cubit.search(todos.first.description),
+      expect: () => [
+        TodoLoadedState([todos.first]),
+      ],
+    );
+
+    blocTest(
+      'cubit emits TodoEmptyState when search with unknown keyword',
+      setUp: () => mockTodoBoxResult(results: todos),
+      build: () => sl<TodoCubit>(),
+      act: (TodoCubit cubit) => cubit.search('unknown'),
+      expect: () => [
+        TodoEmptyState(),
+      ],
+    );
+  });
+
+  group('todo operations', () {
+    List<Todo> todos = [
+      Todo.create(1, 'Pray'),
+      Todo.create(2, 'Eat'),
+      Todo.create(3, 'Work'),
+    ];
+
+    final Todo newTodo = Todo.create(4, 'Sport');
+
+    blocTest(
+      'cubit emits TodoLoadedState with new item after add todo item',
+      setUp: () => mockTodoBoxResult(results: todos),
+      build: () => sl<TodoCubit>(),
+      act: (TodoCubit cubit) => cubit.addTodo(newTodo),
+      expect: () => [
+        TodoLoadedState(todos..add(newTodo)),
+      ],
+    );
+
+    blocTest(
```

Comment:
i can't use expect here, since i don't know an easy way to compare arrays
so i just use verify to check the last item of todos list is updated